deployment/common/changeset/set_config_mcms_test.go

```diff
@@ -0,0 +1,322 @@
+package changeset_test
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+// setupSetConfigTestEnv deploys all required contracts for the setConfig MCMS contract call.
+func setupSetConfigTestEnv(t *testing.T) deployment.Environment {
+
+	lggr := logger.TestLogger(t)
+	cfg := memory.MemoryEnvironmentConfig{
+		Nodes:  1,
+		Chains: 2,
+	}
+	env := memory.NewMemoryEnvironment(t, lggr, zapcore.DebugLevel, cfg)
+	chainSelector := env.AllChainSelectors()[0]
+
+	config := proposalutils.SingleGroupTimelockConfig(t)
+	// Deploy MCMS and Timelock
+	env, err := commonchangeset.ApplyChangesets(t, env, nil, []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployLinkToken),
+			Config:    []uint64{chainSelector},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployMCMSWithTimelock),
+			Config: map[uint64]types.MCMSWithTimelockConfig{
+				chainSelector: config,
+			},
+		},
+	})
+	require.NoError(t, err)
+	return env
+}
+
+// TestSetConfigMCMS tests the SetConfigMCMS changeset by calling SetConfig and checking the config values.
+func TestSetConfigMCMS(t *testing.T) {
+	t.Parallel()
+	ctx := tests.Context(t)
+
+	env := setupSetConfigTestEnv(t)
+	chainSelector := env.AllChainSelectors()[0]
+	chain := env.Chains[chainSelector]
+	addrs, err := env.ExistingAddresses.AddressesForChain(chainSelector)
+	require.NoError(t, err)
+	require.Len(t, addrs, 6)
+
+	mcmsState, err := commonchangeset.MaybeLoadMCMSWithTimelockChainState(chain, addrs)
+	require.NoError(t, err)
+
+	timelockAddress := mcmsState.Timelock.Address()
+	cfg := proposalutils.SingleGroupMCMS(t)
+	// Add the timelock as a signer to check state changes
+	cfg.Signers = append(cfg.Signers, timelockAddress)
+	cfg.Quorum = 2 // quorum should change to 2 out of 2 signers
```

Comment:
better to set different configs per MCM and assert the right config is set on the right MCM. I think a practical bug would be we mix up the MCM configs (i.e. bypasser gets the proposer config) which would be masked by this test